packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -27,6 +29,56 @@ export type ESBuildTransformResult = Omit<TransformResult, 'map'> & {
   map: SourceMap
 }
 
+type TSConfigJSON = {
+  extends?: string
+  compilerOptions?: {
+    target?: string
+    jsxFactory?: string
+    jsxFragmentFactory?: string
+    useDefineForClassFields?: boolean
+    importsNotUsedAsValues?: 'remove' | 'preserve' | 'error'
+  }
+  [key: string]: any
+}
+type TSCompilerOptions = NonNullable<TSConfigJSON['compilerOptions']>
+
+const tsconfigCache = new Map<string, TSConfigJSON>()
+async function loadTsconfigJsonForFile(
+  filename: string
+): Promise<TSConfigJSON> {
+  const directory = path.dirname(filename)
+
+  const cached = tsconfigCache.get(directory)
+  if (cached) {
+    return cached
+  }
+
+  let configPath = await findTSConfig(directory)
```

Comment:
Should we check the initial `configPath` whether is include in 'process.cwd' ?

Reply:
Would you open an issue about this so it is easier to track its resolution?

Reply:
Oh, yes. I'm just not sure about the problem I met is worth solving. 
I create a typescript project that doesn't have `tsconfig.json` in workspace. And `findTSConfig` is trying to recursively find `tsconfig.json` and it finds a `tsconfig.json` in some parents folder. That `tsconfig.json#target` is `es5`, so when I run `yarn vite`, I get an error for esbuild doesn't support es5, which I knew but doesn't know what causes the error until I look into the code.

Reply:
I think that's from esbuild, not this PR.

Because in this PR I only add check for 4 fields: `jsxFactory`, `jsxFragmentFactory`, `useDefineForClassFields`, and `importsNotUsedAsValues`.